airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -232,14 +244,16 @@ public AirbyteMessage getMessage() {
                                                                              JdbcStateManager stateManager,
                                                                              Instant emittedAt) {
     if (isCdc(config)) {
-      final Predicate<ChangeEvent<String, String>> hasReachedLsnPredicate = getTerminationPredicate(database);
+
+      final Predicate<ChangeEvent<String, String>> hasReachedLsnPredicate = getTerminationPredicate(database, stateManager);
       AtomicReference<Throwable> thrownError = new AtomicReference<>();
       AtomicBoolean completed = new AtomicBoolean(false);
       ExecutorService executor = Executors.newSingleThreadExecutor();
       final CloseableLinkedBlockingQueue<DebeziumPayload> queue = new CloseableLinkedBlockingQueue<>(executor::shutdown);
 
       DebeziumEngine<ChangeEvent<String, String>> engine = DebeziumEngine.create(Json.class)
           .using(getDebeziumProperties(config, catalog))
+          .using(new AlwaysCommitOffsetPolicy())
```

Comment:
Probably worth a comment explaining this

Reply:
better yet, i'll kill it. everything gets committed and flushed to the offset file when the engine closes so this shouldn't be needed. we can bring it back when we start doing check pointing.